bluetooth/src/main/java/com/caneseeproject/bluetooth/BluetoothSensorPortal.kt

```diff
@@ -4,22 +4,46 @@ import android.bluetooth.BluetoothAdapter
 import android.bluetooth.BluetoothDevice
 import android.bluetooth.BluetoothSocket
 import com.caneseeproject.sensorPortals.Sensor
+import com.caneseeproject.sensorPortals.SensorControl
 import com.caneseeproject.sensorPortals.SensorPortal
+import com.caneseeproject.sensorPortals.SensorReading
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import java.io.InputStream
+import java.io.OutputStreamWriter
 import java.util.*
 
 
-internal class BluetoothSensorPortal(private val MAC: String) : SensorPortal {
-
+internal class BluetoothSensorPortal(private val MAC: String ) : SensorPortal {
     private val bluetoothAdapter: BluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
     // val HC05 : String = "98:D3:61:FD:66:FB"
     private var device: BluetoothDevice = bluetoothAdapter.getRemoteDevice(MAC)
+    val socket: BluetoothSocket = device.createRfcommSocketToServiceRecord(SERIAL_UUID)
+    private val mmInStream: InputStream = socket.inputStream
```

Comment:
call `createRfcommSocketToServiceRecord` in `open`.
You can make `socket` a global `lateinit var`.